### PR TITLE
Build mt8195 mt8186 into SOF docker image

### PR DESCRIPTION
**Pull request by tinghan-shen**
test test

Signed-off-by: Tinghan Shen <tinghan.shen@mediatek.com>

### CODE REVIEW

**Comment by marc-hb** [comment]
> The CI build for mt8195 and mt8186 is still failed with this PR.

I don't understand. First this PR needs to be merged, then someone needs to rebuild and publish a new Docker image. Only then can the toolchain be used in other PRs. The Docker update process is all manual for now, see discussion in https://github.com/thesofproject/sof/issues/5073 We don't update the docker image very often so it has not been worth the effort to fully automate its creation and test.

Did you build a new docker image on your system with this change? Then did you test it that docker image, I mean did you try to build SOF with it?

**Comment by tinghan-shen** [comment]
Oh, I get your point. I had a miss understanding of the process when I test the CI this PR.
I was a bit worry about the issue of gcc-build-only for mt8186 and mt8195.
I was thought that this has to be solved ASAP.

I guess I can close this PR and wait until the Docker image updated.
I will take some time to build the docker image locally.
thank you, marc.

**Comment by afq984** [comment]
Hi @marc-hb I wonder what is missing for a docker image that supports targeting mt81xx?
The work for xtensa-overlay and crosstool-ng seems complete [1][2].

[1] https://github.com/thesofproject/xtensa-overlay/commit/ccd9727dabf807e58dc48b6bc7efb536334c9244
[2] https://github.com/thesofproject/crosstool-ng/commit/3d5c8ed89c7f60d7d41f8750192ddcda9ec9b969

**Comment by marc-hb** [comment]
@fredoh9 can you try this?

**Comment by marc-hb** [comment]
> I will take some time to build the docker image locally.

Just trying to make sure there is no misunderstanding: @afq984 you did test this PR on your personal system, correct? I mean you created a Docker image with the mt818x toolchains and tested them and they worked fine?

Manual and local testing is required because we have no automated testing for Docker changes (#5073).

I'm asking because you closed this PR on Feb 17 after writing this, I wonder why.

**Comment by afq984** [comment]
> you did test this PR on your personal system, correct? 
no, I ran into some weird ulimit issues to build the image that I hadn't figure out

> I'm asking because you closed this PR on Feb 17
It was @tinghan-shen, not me :)

**Comment by marc-hb** [comment]
Some ulimit defaults are often too small for crosstool-ng

https://github.com/crosstool-ng/crosstool-ng/issues?q=ulimit

Just trying increasing the failing ones temporarily in the terminal before cleaning and trying to build again

**Comment by tinghan-shen** [comment]
> > I will take some time to build the docker image locally.
> 
> I'm asking because ~you~ @tinghan-shen closed this PR on Feb 17 after writing this, I wonder why. Also, it was always a draft, never promoted.

I'm sorry. I haven't done it since then. I'll find some time to do it.
This PR was created to test the CI for mt8195 and mt8186, so I didn't plan to promote this PR.

File: scripts/docker-run.sh [hunk 1]
```diff
@@ -31,6 +31,7 @@ test "$(id -n)" = 1001 ||
     >&2 printf "Warning: this script should be run as user ID 1001 to match the container\n"
 
 set -x
+echo SOF_DOCKER_RUN: $SOF_DOCKER_RUN
```
Comment (marc-hb) [inline scripts/docker-run.sh:34]: The `set -x` should print the command below which contains $SOF_DOCKER_RUN

File: scripts/docker-run.sh [hunk 2]
```diff
@@ -42,4 +43,4 @@ docker run -i -v "${SOF_TOP}":/home/sof/work/sof.git \
 	--env https_proxy="$https_proxy" \
 	--user "$(id -u)" \
 	$SOF_DOCKER_RUN \
-	thesofproject/sof "$@"
+	sof "$@"
```
Comment (marc-hb) [inline scripts/docker-run.sh:46]: Our documentation used to recommend creating such a local `sof` tag but I removed it after wasting too much time forgetting to update it.

**Comment by marc-hb** [comment]
@tinghan-shen I wonder what you're expecting with your recent, "debug" pushes... did you misunderstand or forget this below?

> someone needs to rebuild and publish a new Docker image. Only then can the toolchain be used in other PRs. The Docker update process is all manual for now, see discussion in https://github.com/thesofproject/sof/issues/5073 We don't update the docker image very often so it has not been worth the effort to fully automate its creation and test.

Changes to `scripts/docker_build/sof_builder/Dockerfile` are NOT tested in any CI, every CI uses the static Docker image that was manually built, tested and uploaded.

**Comment by tinghan-shen** [comment]
> @tinghan-shen I wonder what you're expecting with your recent, "debug" pushes... did you misunderstand or forget this below?
> 
> > someone needs to rebuild and publish a new Docker image. Only then can the toolchain be used in other PRs. The Docker update process is all manual for now, see discussion in #5073 We don't update the docker image very often so it has not been worth the effort to fully automate its creation and test.
> 
> Changes to `scripts/docker_build/sof_builder/Dockerfile` are NOT tested in any CI, every CI uses the static Docker image that was manually built, tested and uploaded.

Hi Marc,

I had a local build SOF docker image and still under testing.
I forget to do it in another local branch, but update to this PR branch directly.
Sorry for causing any inconvenience.

File: scripts/docker-run.sh (hunk 1 above)
Comment (tinghan-shen) [reply scripts/docker-run.sh]: This is a hack and dropped now.

**Review by marc-hb — DISMISSED** [review]
Great news, thanks! Let's merge this and yes someone at Intel will update the Docker image (I can't promise when exactly)

**Comment by afq984** [comment]
thanks for doing this :)

**Comment by tinghan-shen** [comment]
> Great news, thanks! Let's merge this and yes someone at Intel will update the Docker image (I can't promise when exactly)

Should I remove the change of .github before the Docker image updated?

**Comment by marc-hb** [comment]
> Should I remove the change for .github before the Docker image updated?

Yes please. It requires an updated Docker image.

You can comment it out and leave a TODO.

**Comment by tinghan-shen** [comment]
> Yes please. It requires an updated Docker image.
> 
> You can comment it out and leave a TODO.

Updated. Thanks!

**Review by marc-hb — APPROVED** [review]
ICL-only Copy build to DUT FAIL in https://sof-ci.01.org/sof-pr-viewer/#/build/PR5388/build9383529, @wszypelt any idea?

Unfortunately common suspend/resume failures in https://sof-ci.01.org/sofpr/PR5388/build430/devicetest, unrelated.

**Comment by marc-hb** [comment]
> SOFCI TEST

`scripts/docker_build/sof_builder/Dockerfile` is not tested in CI, see discussion above.

**Pull request merged**